### PR TITLE
Feat: Enable proxy-authorization in admin client

### PR DESCRIPTION
**Pull request by fg91**
# TL;DR

Part of an effort to integrate Flyte with [GCP Identity Aware Proxy](https://cloud.google.com/iap), see https://github.com/flyteorg/flyte/issues/3965.

Allows the flyteidl admin client to sent a `"proxy-authorization"` header with every request to flyteadmin. Tokens for this header are created using an external command which is configured via the new `proxyCommand` config entry.

Same logic as was introduced in `flytekit` in https://github.com/flyteorg/flytekit/pull/1787.

I replicated the existing logic in `MaterializeCredentials` and `NewAuthInterceptor` for the second token. The external command is called a single time per `flytectl` call.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
See https://github.com/flyteorg/flyte/issues/3965

## Tracking Issue

Closes https://github.com/flyteorg/flyte/issues/3965

## Follow-up issue
_NA_

### CODE REVIEW

File: clients/go/admin/auth_interceptor.go [hunk 1]
```diff
@@ -48,19 +51,70 @@ func MaterializeCredentials(ctx context.Context, cfg *Config, tokenCache cache.T
...
+		return nil, fmt.Errorf("failed to initialized proxy authorization token source provider. Err: %w", err)
+	}
+	proxyTokenSource, err := tokenSourceProvider.GetTokenSource(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return proxyTokenSource, nil
+}
+
+func MaterializeProxyAuthCredentials(ctx context.Context, cfg *Config, proxyCredentialsFuture *PerRPCCredentialsFuture) error {
```
Comment (fg91) [inline clients/go/admin/auth_interceptor.go:66]: Adapted from `MaterializeAuthCredentials`.

File: clients/go/admin/auth_interceptor.go [hunk 2]
```diff
@@ -102,3 +156,18 @@ func NewAuthInterceptor(cfg *Config, tokenCache cache.TokenCache, credentialsFut
 		return err
 	}
 }
+
```
Comment (fg91) [inline clients/go/admin/auth_interceptor.go:159]: Adapted from `NewAuthInterceptor`.

File: clients/go/admin/pkce/handle_app_call_back.go [hunk 3]
```diff
@@ -43,7 +43,8 @@ func getAuthServerCallbackHandler(c *oauth.Config, codeVerifier string, tokenCha
 		var opts []oauth2.AuthCodeOption
 		opts = append(opts, oauth2.SetAuthURLParam("code_verifier", codeVerifier))
 
-		token, err := c.Exchange(context.Background(), req.URL.Query().Get("code"), opts...)
+		ctx := context.WithValue(context.Background(), oauth2.HTTPClient, client)
+		token, err := c.Exchange(ctx, req.URL.Query().Get("code"), opts...)
 		if err != nil {
```
Comment (fg91) [inline clients/go/admin/pkce/handle_app_call_back.go:48]: If using a proxy, we need to use the http client created in `auth_interceptor.go` to retrieve the token. So far, this client wasn't used.

File: clients/go/admin/auth_interceptor.go [hunk 4]
```diff
@@ -102,3 +156,18 @@ func NewAuthInterceptor(cfg *Config, tokenCache cache.TokenCache, credentialsFut
...
+func NewProxyAuthInterceptor(cfg *Config, proxyCredentialsFuture *PerRPCCredentialsFuture) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+
+		err := invoker(ctx, method, req, reply, cc, opts...)
+		if err != nil {
+			newErr := MaterializeProxyAuthCredentials(ctx, cfg, proxyCredentialsFuture)
```
Comment (fg91) [inline clients/go/admin/auth_interceptor.go:165]: The external command is called a single time per call of `flytectl`. After the call of the external command, the credentials future is initialized and can be reused for further grpc calls.

**Comment by eapolinario** [comment]
I moved this PR to the monorepo: https://github.com/flyteorg/flyte/pull/4189.
